### PR TITLE
Add product-specification condition

### PR DESCRIPTION
**Pull request by LucasCastroAcctGlobal**
#### What is the purpose of this pull request?

Add a new condition with a product specification filter.

#### What problem is this solving?

Add a new property where the condition meets the past product specification.

### How should this be manually tested?
- Enter this link: https://condition--tackoftheday.myvtex.com/elliena-eq-kids-the-edge-polo-470453-greygreystripe/p
- The "Size-Chart" tab-item will not appear because the product does not have this specification;
- Enter this link: https://condition--tackoftheday.myvtex.com/horseware-womens-softie-socks-triangle-print-cohfcr-8tri/p
- Now the "Size-Chart" tab-item will appear because the product has this "Size Chart:" specification.

### Screenshots or example usage
https://cl.ly/487487800767

Types of changes
- [ ] - Bug fix (a non-breaking change which fixes an issue)
- [X] - New feature (a non-breaking change which adds functionality)
- [ ] - Breaking change (fix or feature that would cause existing functionality to change)
- [ ] - Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by victorhmp** [comment]
Hi @LucasCastroAcctGlobal !
First of all, thanks for taking the time to contribute with this awesome PR, it looks really good! 😄 

However, I do have a suggestion. Why don't we change the name of the new prop to `"productSpecifications"`? This seems more in line with the naming used for the existing props.

Also, I've noticed there are two checks that are failing, the `IO app test` and `Lint` checks. Would you mind fixing those? For the `Lint` one, a simple `yarn lint --fix` should do the trick :)

File: docs/README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ The `condition-layout.product` block provides the subjects below to use in `cond
 | `selectedItemId`  | `value` | Id of the current selected SKU |
 | `productClusters` | `array` | List of product clusters.             |
 | `categoryTree`    | `array` | List of categories.                   |
+| `properties`      | `array` | List of product-specifications.       |
```
Comment (kaisermann) [inline docs/README.md:53]: I think `specifications` would be a better fit here. What do you think?

Comment (kaisermann) [reply docs/README.md:53]: Or `specificationProperties`

Comment (LucasCastroAcctGlobal) [reply docs/README.md:53]: Hello @kaisermann and @victorhmp!

If possible, I wanted your help on two points.

First, I am having problems with the App IO test, and in my development branch the test is OK.

And second, I'm having trouble changing the name of the prop from "properties" to "specificationProducts", as you suggested. Could you help me how can I do that, please?

Best Regards!

Comment (kaisermann) [reply docs/README.md:53]: Hey @LucasCastroAcctGlobal 👋  The tests seem to be passing now 🤷. About the prop name, you could rename it when destructuring the `product` context value and then passing the value, with the new name, to the `useMemo`:

```js
 const {
    productId,
    categoryId,
    brandId,
    productClusters,
    categoryTree,
    properties: specificationProperties,
  } = product ?? {}
  const { itemId: selectedItemId } = selectedItem ?? {}

  // We use a useMemo to modify the a condition layout "values"
  // only when some of the context props change.
  const values = useMemo(
    () => ({
      productId,
      categoryId,
      brandId,
      productClusters,
      categoryTree,
      selectedItemId,
      specificationProperties,
    }),
    [
      brandId,
      categoryId,
      categoryTree,
      productClusters,
      productId,
      selectedItemId,
      specificationProperties,
    ]
  )```

Comment (LucasCastroAcctGlobal) [reply docs/README.md]: Hello @kaisermann!

Thanks a lot for the help! I made the modifications and tests and it worked.

Let me know if there is anything else to fix!

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- New prop "properties" enabling one more condition: by product specification.
```
Comment (victorhmp) [inline CHANGELOG.md:11]:
```suggestion
- New `specificationProperties ` enabling one more condition: by product specification.
```

**Comment by LucasCastroAcctGlobal** [comment]
@guerreirobeatriz after committing your suggestions, github is showing a branches conflict. Can you fix this?

**Comment by victorhmp** [comment]
> @guerreirobeatriz after committing your suggestions, github is showing a branches conflict. Can you fix this?

Sorry, but I don't think that the conflict is only because of changes in the README, but also the in the CHANGELOG, and @guerreirobeatriz should not need to rebase this.

**Comment by klzns** [comment]
@LucasCastroAcctGlobal I think you ran `git pull` and not `git rebase master`. I can fix that but I don't have permissions to push to your branch.

I'll leave the instructions on how to if you are feeling like doing it:


```sh
git remote add upstream git@github.com:vtex-apps/condition-layout.git
git fetch upstream
git checkout NOMEDABRANCH
git rebase upstream/master
# resolver os conflitos
git push --force
```

We plan to publish it on Monday.

**Comment by LucasCastroAcctGlobal** [comment]
> @LucasCastroAcctGlobal I think you ran `git pull` and not `git rebase master`. I can fix that but I don't have permissions to push to your branch.
> 
> I'll leave the instructions on how to if you are feeling like doing it:
> 
> ```shell
> git remote add upstream git@github.com:vtex-apps/condition-layout.git
> git fetch upstream
> git checkout NOMEDABRANCH
> git rebase upstream/master
> # resolve conflicts
> git push --force
> ```
> 
> We plan to publish it on Monday.

@klzns The problems have already been fixed. Could you validate if anything else is missing, please?

**Comment by klzns** [comment]
Thanks! Merged at #18

**Comment by klzns** [comment]
@allcontributors add @LucasCastroAcctGlobal for code

**Pull request closed without merging**